src/lib/sql/sql_translator.cpp

```diff
@@ -1323,6 +1300,65 @@ std::shared_ptr<AbstractExpression> SQLTranslator::_translate_hsql_case(
   return current_case_expression;
 }
 
+std::shared_ptr<AbstractExpression> SQLTranslator::_inverse_predicate(const AbstractExpression& expression) const {
+  /**
+   * Inverse a boolean expression
+   */
+
+  switch (expression.type) {
+    case ExpressionType::Predicate: {
+      if (const auto* binary_predicate_expression = dynamic_cast<const BinaryPredicateExpression*>(&expression);
+          binary_predicate_expression) {
+        // If the argument is a predicate, just inverse it (e.g. NOT (a > b) becomes b <= a)
+        return std::make_shared<BinaryPredicateExpression>(
+            inverse_predicate_condition(binary_predicate_expression->predicate_condition),
+            binary_predicate_expression->left_operand(), binary_predicate_expression->right_operand());
+      } else if (const auto is_null_expression = dynamic_cast<const IsNullExpression*>(&expression);
+                 is_null_expression) {
+        // NOT (IS NULL ...) -> IS NOT NULL ...
+        return std::make_shared<IsNullExpression>(inverse_predicate_condition(is_null_expression->predicate_condition),
+                                                  is_null_expression->operand());
+      } else if (const auto* between_expression = dynamic_cast<const BetweenExpression*>(&expression);
+                 between_expression) {
+        // a BETWEEN b AND c -> a < b OR a > c
+        return or_(less_than_(between_expression->value(), between_expression->lower_bound()),
+                   greater_than_(between_expression->value(), between_expression->upper_bound()));
+      } else {
+        const auto* in_expression = dynamic_cast<const InExpression*>(&expression);
```

Comment:
This should have an assertion to make sure that there is no case that we have left out